src/topology/uniform_space/uniform_convergence.lean

```diff
@@ -179,6 +188,52 @@ calc tendsto ↿F (p ×ᶠ ⊤) (𝓝 c)
 ... ↔ ∀ V ∈ 𝓤 β, {x | (c, ↿F x) ∈ V} ∈ p ×ᶠ (⊤ : filter α) : iff.rfl
 ... ↔ ∀ V ∈ 𝓤 β, {i | ∀ a, (c, F i a) ∈ V} ∈ p : by simpa [mem_prod_top]
 
+/-- Uniform convergence on the empty set is vacuously true -/
+lemma tendsto_uniformly_on_of_empty :
+  tendsto_uniformly_on F f p ∅ :=
+λ u hu, by simp
+
+/-- Uniform convergence on a singleton is equivalent to regular convergence -/
+lemma tendsto_uniformly_on_singleton_iff_tendsto :
+  tendsto_uniformly_on F f p {x} ↔ tendsto (λ n : ι, F n x) p (𝓝 (f x)) :=
+begin
+  rw uniform.tendsto_nhds_right,
+  unfold tendsto,
+  rw filter.le_def,
+  simp_rw filter.mem_map',
+
+  split,
+  exact (λ h u hu, by simpa using eventually_iff.mp (h u hu)),
+  exact (λ h u hu, by simpa using eventually_iff.mp (h u hu)),
+end
+
+lemma filter.tendsto.tendsto_uniformly_on_const
+  {g : ι → β} {b : β} (hg : tendsto g p (𝓝 b)) (s : set α) :
+  tendsto_uniformly_on (λ n : ι, λ a : α, g n) (λ a : α, b) p s :=
```

Comment:
Is there a version of `tendsto_prod_top_iff` for `tendsto_uniformly`? Maybe with `\top` replaced by `𝓟 s`?
I expect that such a lemma can be used to prove this lemma (and maybe even turn it into an `iff`?

Reply:
Not to my knowledge. But I can take a look. You're right that it should be something very similar.

Reply:
You don't actually need to use that `s` is nonempty in your proof, since you only need the implication right-to-left.

Reply:
:+1: